src/main/java/net/neoforged/neoforge/network/registration/NetworkRegistry.java

```diff
@@ -135,19 +135,27 @@ public static void setup() {
     /**
      * Registers a new payload.
      * 
-     * @param <T>       The class of the payload.
-     * @param <B>       The class of the ByteBuf. Only {@link ConnectionProtocol#PLAY play} payloads may use {@link RegistryFriendlyByteBuf}.
-     * @param type      The type of the payload.
-     * @param codec     The codec for the payload.
-     * @param handler   The handler for the payload. This handler should expect to receive the payload on all declared protocols and flows. It will be executed on the network thread.
-     * @param protocols The protocols this payload supports being sent over. Only {@link ConnectionProtocol#CONFIGURATION configuration} and {@link ConnectionProtocol#PLAY play} are supported.
-     * @param flow      The flow of this payload. Specify {@link Optional#empty()} to support sending in both directions.
-     * @param version   The version of the payload. Increase the payload version if the codec logic or handler logic changes. Neo-Neo connections with mismatched versions are denied.
-     * @param optional  If the payload is optional. Any connection with missing non-optional payloads is denied.
+     * @param <T>           The class of the payload.
+     * @param <B>           The class of the ByteBuf. Only {@link ConnectionProtocol#PLAY play} payloads may use {@link RegistryFriendlyByteBuf}.
+     * @param type          The type of the payload.
+     * @param codec         The codec for the payload.
+     * @param serverHandler The server-side handler for the payload. This handler will be executed on the network thread.
+     * @param clientHandler The client-side handler for the payload. This handler will be executed on the network thread.
+     * @param protocols     The protocols this payload supports being sent over. Only {@link ConnectionProtocol#CONFIGURATION configuration} and {@link ConnectionProtocol#PLAY play} are supported.
+     * @param flow          The flow of this payload. Specify {@link Optional#empty()} to support sending in both directions.
+     * @param version       The version of the payload. Increase the payload version if the codec logic or handler logic changes. Neo-Neo connections with mismatched versions are denied.
+     * @param optional      If the payload is optional. Any connection with missing non-optional payloads is denied.
      */
     @SuppressWarnings({ "unchecked", "rawtypes" })
-    public static <T extends CustomPacketPayload, B extends FriendlyByteBuf> void register(CustomPacketPayload.Type<T> type, StreamCodec<? super B, T> codec, IPayloadHandler<T> handler,
-            List<ConnectionProtocol> protocols, Optional<PacketFlow> flow, String version, boolean optional) {
+    public static <T extends CustomPacketPayload, B extends FriendlyByteBuf> void register(
+            CustomPacketPayload.Type<T> type,
+            StreamCodec<? super B, T> codec,
+            @Nullable IPayloadHandler<T> serverHandler,
+            @Nullable IPayloadHandler<T> clientHandler,
```

Comment:
In the spirit of having them be separate, I think we should push the client handlers entirely to the client network registry and not have them be registerable through this method.

Reply:
I disagree with this statement.
I and many others have this working 100% properly.

I don't think this is needed at all, and is just a chore for modders to update to.

We should however update our recommendation on the docs.